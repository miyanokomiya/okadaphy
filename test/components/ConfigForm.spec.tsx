import * as React from 'react'
import renderer from 'react-test-renderer'
import OptionForm, { Props } from '../../src/components/ConfigForm'
import { mockConfig } from '../mock'

jest.mock('@material-ui/core/Slider', () => 'Slider')
jest.mock('@material-ui/core/TextField', () => 'TextField')
jest.mock('react-color', () => ({ CompactPicker: 'CompactPicker' }))

describe('src/components/ConfigForm', () => {
  it('snapshot', () => {
    const props: Props = {
      config: mockConfig(),
      onSubmit: () => {},
    }
    const tree = renderer.create(<OptionForm {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
