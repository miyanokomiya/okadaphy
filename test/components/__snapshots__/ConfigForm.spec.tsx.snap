// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/ConfigForm snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Reset
        </span>
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container"
    style={
      Object {
        "marginTop": "1rem",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Text
      </p>
      <TextField
        fullWidth={true}
        multiline={true}
        onChange={[Function]}
        value="mock text"
        variant="outlined"
      />
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    style={
      Object {
        "marginTop": "1rem",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Fill
      </p>
      <CompactPicker
        color="mock fill style"
        onChange={[Function]}
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Stroke
      </p>
      <CompactPicker
        color="mock stroke style"
        onChange={[Function]}
      />
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    style={
      Object {
        "marginTop": "1rem",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Gravity X
      </p>
      <Slider
        marks={
          Array [
            Object {
              "label": "0",
              "value": 0,
            },
          ]
        }
        max={0.05}
        min={-0.05}
        onChange={[Function]}
        step={0.0001}
        value={-1}
        valueLabelFormat={[Function]}
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Gravity Y
      </p>
      <Slider
        marks={
          Array [
            Object {
              "label": "0",
              "value": 0,
            },
          ]
        }
        max={0.05}
        min={-0.05}
        onChange={[Function]}
        step={0.0001}
        value={1}
        valueLabelFormat={[Function]}
      />
    </div>
  </div>
</form>
`;
